pallets/settlement/src/lib.rs

```diff
@@ -2273,6 +2334,38 @@ impl<T: Config> Module<T> {
         Ok(())
     }
 
+    /// Ensures that all tickers in the instruction that have venue filtering enabled are also
+    /// in the venue allowed list.
+    fn ensure_allowed_venue(
+        instruction_legs: &[(LegId, LegV2)],
+        venue_id: VenueId,
+        weight_meter: &mut WeightMeter,
+    ) -> DispatchResult {
+        // Avoids reading the storage multiple times for the same ticker
+        let mut tickers: BTreeSet<Ticker> = BTreeSet::new();
+
+        for (_, leg) in instruction_legs {
+            let ticker = leg.asset.ticker_and_amount().0;
+            if let Err(e) = Self::ensure_venue_filtering(&mut tickers, ticker, &venue_id) {
+                // Charges the weight consumed
+                weight_meter.check_accrue(<T as Config>::WeightInfo::ensure_allowed_venue(
+                    tickers.len() as u32,
+                ));
+                Self::deposit_event(RawEvent::VenueUnauthorized(
+                    SettlementDID.as_id(),
+                    ticker,
+                    venue_id,
+                ));
+                return Err(e);
```

Comment:
We shouldn't be emitting this event, since throwing an error will revert the transaction and events will be reverted too.

Reply:
This is an old issue, just noticed it now.